store/iscn.ts

```diff
@@ -86,57 +86,74 @@ export default class ISCN extends VuexModule {
   @Action
   async queryISCNByAddress(address: string): Promise<ISCNRecordWithID[]> {
     this.context.commit('clearRecords');
-    const client = await getQueryClient();
-    let res;
-    let nextSequence;
     let records: ISCNRecord[] = [];
-    do {
-      this.context.commit('setIsLoading', true);
-      // eslint-disable-next-line no-await-in-loop
-      res = await client.queryRecordsByOwner(address, nextSequence).catch(() => { })
-      if (res) {
-        nextSequence = res.nextSequence.toNumber();
-        this.context.commit('appendRecords', res.records)
-        records = records.concat(res.records);
-      }
-    } while (nextSequence !== 0);
-    this.context.commit('setIsLoading', false);
-    return addIDToRecords(records);
+    try {
+      const client = await getQueryClient()
+      let res
+      let nextSequence
+      do {
+        this.context.commit('setIsLoading', true)
+        // eslint-disable-next-line no-await-in-loop
+        res = await client
+          .queryRecordsByOwner(address, nextSequence)
+          .catch(() => {})
+        if (res) {
+          nextSequence = res.nextSequence.toNumber()
+          this.context.commit('appendRecords', res.records)
+          records = records.concat(res.records)
+        }
+      } while (nextSequence !== 0)
+    } catch (error) {
+      console.error(error)
```

Comment:
Do we want to throw the error for caller to handle error instead?

Reply:
Add an error handler to open a warning snackbar in page.